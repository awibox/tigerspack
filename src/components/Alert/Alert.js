import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { colors } from '../../utils/colors';

const Alert = (props) => {
  const {
    children,
    dismissible,
    outline,
    theme,
    closeIconSize,
    padding,
    ...otherProps
  } = props;
  const palette = colors[theme] ? colors[theme] : colors.primary;
  const styles = {
    alert: {
      padding: `${padding}px`,
      background: outline ? palette.hoverColor : palette.color,
      border: `1px solid ${palette.color}`,
      color: palette.text,
      boxShadow: '0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24)',
      transition: 'all 0.3s cubic-bezier(.25, .8, .25, 1)',
      borderRadius: '4px',
      marginBottom: `${padding}px`,
      position: 'relative',
      width: '100%',
    },
    close: {
      textAlign: 'left',
      position: 'absolute',
      transform: 'rotate(45deg)',
      cursor: 'pointer',
      fontFamily: '-webkit-pictograph',
      fontWeight: 'bold',
      lineHeight: `${closeIconSize}px`,
      width: `${closeIconSize}px`,
      height: `${closeIconSize}px`,
      fontSize: `${closeIconSize}px`,
      right: `${padding / 2}px`,
      top: `${padding}px`,
      userSelect: 'none',
      opacity: 0.7,
      ':hover': {
        opacity: 1,
      },
    },
  };
  const [alertState, setAlertState] = useState(false);
  return (
    <div css={styles.alert} style={{ display: alertState ? 'none' : 'block' }} {...otherProps}>
      {dismissible ? <div css={styles.close} onClick={() => setAlertState(true)}>+</div> : ''}
      {children}
    </div>
  );
};

Alert.propTypes = {
  theme: PropTypes.string,
  children: PropTypes.any.isRequired,
  dismissible: PropTypes.bool,
  outline: PropTypes.bool,
  closeIconSize: PropTypes.number,
  padding: PropTypes.number,
};

Alert.defaultProps = {
  closeIconSize: 28,
  padding: 15,
};

export default Alert;
