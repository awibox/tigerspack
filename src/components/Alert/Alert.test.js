import React from 'react';
import { shallow } from 'enzyme';
import Alert from './Alert';

describe('Alert', () => {
  it('Default Alert', () => {
    const component = shallow(<Alert>Test alert</Alert>);
    expect(component).toMatchSnapshot();
  });
  it('Outline Alert', () => {
    const component = shallow(<Alert outline theme={'danger'}>Test alert</Alert>);
    expect(component).toMatchSnapshot();
  });
  it('Dismissible Alert', () => {
    const component = shallow(<Alert dismissible theme={'success'}>Test alert</Alert>);
    expect(component).toMatchSnapshot();
  });
  it('Dismissible Alert close function', () => {
    const component = shallow(<Alert dismissible>Test alert</Alert>);
    component.find(`.${component.get(0).props.children[0].props.className}`).simulate('click');
    expect(component.get(0).props.style.display).toEqual('none');
  });
});
