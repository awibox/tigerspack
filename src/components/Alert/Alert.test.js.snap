// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Default Alert 1`] = `
.alert {
  padding: 15px !important;
  background: #3F88C5 !important;
  border: 1px solid #3F88C5 !important;
  color: #fff !important;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24) !important;
  -webkit-transition: all 0.3s cubic-bezier(.25, .8, .25, 1) !important;
  -moz-transition: all 0.3s cubic-bezier(.25, .8, .25, 1) !important;
  transition: all 0.3s cubic-bezier(.25, .8, .25, 1) !important;
  border-radius: 4px !important;
  margin-bottom: 15px !important;
  position: relative !important;
}

<div
  className="alert"
  style={
    Object {
      "display": "block",
    }
  }
>
  Test alert
</div>
`;

exports[`Alert Dismissible Alert 1`] = `
.alert {
  padding: 15px !important;
  background: #9BC53D !important;
  border: 1px solid #9BC53D !important;
  color: #fff !important;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24) !important;
  -webkit-transition: all 0.3s cubic-bezier(.25, .8, .25, 1) !important;
  -moz-transition: all 0.3s cubic-bezier(.25, .8, .25, 1) !important;
  transition: all 0.3s cubic-bezier(.25, .8, .25, 1) !important;
  border-radius: 4px !important;
  margin-bottom: 15px !important;
  position: relative !important;
}

.close {
  text-align: left !important;
  position: absolute !important;
  -webkit-transform: rotate(45deg) !important;
  -ms-transform: rotate(45deg) !important;
  transform: rotate(45deg) !important;
  cursor: pointer !important;
  font-family: -webkit-pictograph !important;
  font-weight: bold !important;
  line-height: 28px !important;
  width: 28px !important;
  height: 28px !important;
  font-size: 28px !important;
  right: 7.5px !important;
  top: 15px !important;
  -webkit-user-select: none !important;
  -moz-user-select: none !important;
  -ms-user-select: none !important;
  user-select: none !important;
  opacity: 0.7 !important;
}

.close:hover {
  opacity: 1 !important;
}

<div
  className="alert"
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    className="close"
    onClick={[Function]}
  >
    +
  </div>
  Test alert
</div>
`;

exports[`Alert Outline Alert 1`] = `
.alert {
  padding: 15px !important;
  background: #E76846 !important;
  border: 1px solid #D15130 !important;
  color: #fff !important;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24) !important;
  -webkit-transition: all 0.3s cubic-bezier(.25, .8, .25, 1) !important;
  -moz-transition: all 0.3s cubic-bezier(.25, .8, .25, 1) !important;
  transition: all 0.3s cubic-bezier(.25, .8, .25, 1) !important;
  border-radius: 4px !important;
  margin-bottom: 15px !important;
  position: relative !important;
}

<div
  className="alert"
  style={
    Object {
      "display": "block",
    }
  }
>
  Test alert
</div>
`;
